src/App.js

```diff
@@ -4,25 +4,52 @@ import Header from "./Header";
 import Card from "./Card";
 import data from "./resources/data.json";
 import SearchBar from "./SearchBar";
-import { useState } from "react";
+import { useState, useEffect } from "react";
+import NoResultsFound from "./NoResultsFound";
 
 const CardContainer = ({ filteredMembers }) => {
+  if (!filteredMembers.length) return <NoResultsFound />;
+
   const cards = filteredMembers.map((e) => {
     return <Card member={e} key={e.id} className="card" />;
   });
   return cards;
 };
 
 const App = () => {
-  const [filteredMembers, setFilteredMembers] = useState(data);
+  const [listOfMembers, setListOfMembers] = useState([]);
+  const [filteredMembers, setFilteredMembers] = useState([]);
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  async function fetchData() {
+    const detailsOfUsers = [];
+    for (let user in data) {
+      const user_data = await fetch(
+        `https://api.github.com/users/${data[user].username}`
+      );
+      const json = await user_data.json();
+      detailsOfUsers.push(json);
+    }
+    setListOfMembers(detailsOfUsers);
+  }
 
   return (
     <div id="heading" className="title">
       <Header />
       <div className="body">
-        <SearchBar setFilteredMembers={setFilteredMembers} />
+        <SearchBar
+          listOfMembers={listOfMembers}
+          setFilteredMembers={setFilteredMembers}
+        />
         <div className="row">
-          <CardContainer filteredMembers={filteredMembers} />
+          <CardContainer
+            filteredMembers={
+              filteredMembers.length ? filteredMembers : listOfMembers
```

Comment:
If you search for any wrong query, your code is showing the listOfMembers instead of showing noResdultsFound.

Reply:
Fixed!